### PR TITLE
Documentation structure and heading pages

### PR DESCRIPTION
**Pull request by piotrm0**
- Organized documentation into folders.
- Added "heading" pages for each section, these are in `index.md` files except for the top level one in `docs.md` because the index name is already taken for the intro page.
- Some heading pages are placeholders and temporarily commented out from showing up in the navigation.
- Some pages such as "base" or "overview" were made heading pages. This reduces the number of items in the navigation.
- Added a heading page for the getting started section to serve as the new initial page for the documentation. This page includes the content of the installation page as well as pointers to the other documentation sections.
- Moved `TECHDEBT.md` and `CONTRIBUTING.md` from their old locations and into docs under the contributing section as `techdebt.md` and a heading page`index.md`, respectively.
- Removed some duplicate files.

----

| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit 0ec4d2f6841b340d838f88ef1f68dc625f217a1d.  | 
|--------|--------|

### Summary:
This PR restructures the documentation, introduces 'heading' pages, moves `TECHDEBT.md` and `CONTRIBUTING.md` into the docs, and removes duplicate files.

**Key points**:
- Organized documentation into folders.
- Introduced 'heading' pages for each section.
- Moved `TECHDEBT.md` and `CONTRIBUTING.md` into docs under the contributing section.
- Removed some duplicate files.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: docs/trulens_eval/contributing/index.md [hunk 1]
```diff
@@ -67,16 +70,43 @@ All example notebooks are expected to:
...
+### 💣 Tech Debt
+
+Parts of the code are nuanced in ways should be avoided by new contributors.
+Discussions of these points are welcome to help the project rid itself of these
+problematic designs. See [Tech debt index](techdebt).
+
+## Contributors
+
+{%
+   include-markdown "../../../trulens_eval/CONTRIBUTORS.md"
```
[the remark this answers is not in the document]
Comment (piotrm0) [reply docs/trulens_eval/contributing/index.md:104]: @ellipsis-dev CONTRIBUTORS.md are presently not under docs folder but under the python module folders. Your suggestion would point to wrong locations.

File: docs/trulens_eval/contributing/index.md [hunk 2]
```diff
@@ -67,16 +70,27 @@ All example notebooks are expected to:
...
 
 ## 📄 Improve Code Quality & Documentation
+
 We would love your help in making the project cleaner, more robust, and more
 understandable. If you find something confusing, it most likely is for other
 people as well. Help us be better!
 
-## Standards
+Big parts of the code base currently do not follow the code standards outlined
+in [Standards index](standards). Many good contributions can be made in adapting
```
[the remark this answers is not in the document]
Comment (piotrm0) [reply docs/trulens_eval/contributing/index.md:85]: Can you explain why?

[the remark this answers is not in the document]
Comment (piotrm0) [reply docs/trulens_eval/contributing/index.md:85]: @ellipsis-dev can you explain why?

[the remark this answers is not in the document]
Comment (piotrm0) [reply docs/trulens_eval/contributing/index.md:85]: @ellipsis-dev your suggestion of absolute path "/contributing/standards" is incorrect as "contributing" is not at the top level. It is under "trulens_eval".

[the remark this answers is not in the document]
Comment (piotrm0) [reply docs/trulens_eval/contributing/index.md:85]: @ellipsis-dev: can you explain why using absolute paths is preferable to the relative ones?

File: mkdocs.yml [hunk 3]
```diff
@@ -139,113 +142,121 @@ theme:
...
-      - Instrumentation:
-          # Titles set inside notebooks and will be overridden if provider here.
-          - trulens_eval/trulens_instrumentation.ipynb
-          - trulens_eval/langchain_instrumentation.ipynb
-          - trulens_eval/llama_index_instrumentation.ipynb
-      - Logging:
-          - Where to Log: trulens_eval/where_to_log.md
-          - 📓 Logging Methods: trulens_eval/logging.ipynb
+    # PLACEHOLDER: - trulens_eval/tracking/index.md
+    - Instrumentation:
```
Comment (joshreini1) [inline mkdocs.yml:191]: Instrumentation overview is now just accessible from the header which is non-obvious. Can you re-add it as a page under the Instrumentation header?

Screenshot 2024-03-01 at 3 03 29 PM

**Review by joshreini1 — APPROVED** [review]

File: mkdocs.yml (hunk 3 above)
Comment (piotrm0) [reply mkdocs.yml:191]: Instrumentation heading/overview is the first page in the tracking section. It is impossible avoiding getting on this page. Also, part of the point of this PR is to move overviews to the headings. Could rename the page as "Instrumentation Overview" to make it more obvious.

**Pull request merged**